### PR TITLE
Checks if transport device should be periodic

### PR DESCRIPTION
**Pull request by bhourahine**
Also adds some missing comments

### CODE REVIEW

File: src/dftbp/transport/negfvars.F90 [hunk 1]
```diff
@@ -249,6 +249,9 @@ module dftbp_transport_negfvars
     !> type of phonon modes (Longitudinal, transverse)
     integer :: typeModes
 
+    !> Tolerance for contact vectors being distorted
+    real(dp) :: contactlayertol
```
Comment (aradi) [inline src/dftbp/transport/negfvars.F90:253]:
```suggestion
    real(dp) :: contactLayerTol
```

File: src/dftbp/transport/negfint.F90 [hunk 2]
```diff
@@ -135,6 +142,20 @@ subroutine TNegfInt_init(this, transpar, env, greendens, tundos, tempElec, coord
...
+    if (isPeriodic .and. nCont > 2) then
+      allocate(sigma(min(nCont,3)))
+      allocate(contVectors(3, nCont))
+      allocate(U(3, 3))
+      allocate(Vt(nCont, nCont))
+      do iCont = 1, nCont
+        contVectors(:, iCont) = transpar%contacts(iCont)%lattice
+      end do
+      call gesvd(contVectors, U, sigma, Vt)
+      if (all(sigma > transpar%contactLayerTol**2)) then
```
Comment (aradi) [inline src/dftbp/transport/negfint.F90:154]: Could you factor that out into a function ˙hasFullySurroundingContacts` or something similar? Then we would not need all the auxiliary variables in the local scope.

File: src/dftbp/transport/negfvars.F90 (hunk 1 above)
Comment (bhourahine) [reply src/dftbp/transport/negfvars.F90]: Fixed

File: src/dftbp/transport/negfint.F90 (hunk 2 above)
Comment (bhourahine) [reply src/dftbp/transport/negfint.F90]: Sure, done.

**Pull request merged**